client/allocrunner/taskrunner/consul_hook.go

```diff
@@ -56,10 +57,16 @@ func (h *consulHook) Prestart(ctx context.Context, req *interfaces.TaskPrestartR
 
 	// Write tokens to tasks' secret dirs
 	for _, t := range tokens {
-		for identity, token := range t {
+		for tokenName, token := range t {
+			s := strings.Split(tokenName, "/")
```

Comment:
to be pedantic there is https://pkg.go.dev/strings#SplitN though shouldn't be necessary since task names cannot include slashes